File: 2021/test/day_03_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_03"

class Day03Test < Minitest::Test
  def test_part1
    day = Day03.new

    assert_equal 198, day.part1(input("day_03_example_1"))
    assert_equal 3_847_100, day.part1(input("day_03"))
  end

  def test_part2
    day = Day03.new

    assert_equal 230, day.part2(input("day_03_example_1"))
    assert_equal 4_105_235, day.part2(input("day_03"))
  end
end

File: 2025/test/day_03_test.exs
defmodule Day03Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day03.part_1(input("day_03_example_1")) == 357
    assert Day03.part_1(input("day_03")) == 17324
  end

  test "part2" do
    assert Day03.part_2(input("day_03_example_1")) == 3_121_910_778_619
    assert Day03.part_2(input("day_03")) == 171_846_613_143_331
  end
end
